### DISCUSSION TITLE
Using TESPy as a replacement for TRNSYS

### DISCUSSION THREAD

**Discussion by PabloCastilloFCR** [oemof/tespy#541] [General]
Hello everyone!

I've recently started exploring TESPy and am truly impressed by its capabilities. The simple and advanced examples I've seen showcase just how powerful this library can be. Currently, I'm working on a solar energy project that requires an annual evaluation of a solar field with thermal storage and its integration with an industrial process. Normally, I would use TRNSYS for this task, but I'm convinced that TESPy might have the potential to handle it just as well.

I've gone through the forums and found some implementations of thermal storage, but I haven't come across any guidance on how to conduct transient analysis using weather data (TMY), which is a common approach in TRNSYS. I would like to start simulating just a solar collector, a thermal storage, a pump and a constant load. 

Any help or guidance on how to achieve this with TESPy would be greatly appreciated. If I'm able to successfully simulate the plant, I'd be happy to share the results with the community.

**Comment by fwitte** [comment]
Hi @PabloCastilloFCR and thank you for reaching out!

TESPy is not meant to do transient analysis per se, so you have to do some implementations by yourself. What you could do:

- Use the tespy components models and simulate the properties of the fluid leaving the solar collector and exchanging heat to the storage
- Implement some kind of storage model yourself, that calculates change of temperature given the fluid information of the solar collector model
- Implement some kind of control strategy for the storage filling level (when is storage charged, when discharged and when in standstill).

An alternative to using tespy for the transient simulation would be to use dispatch optimization software, e.g. https://github.com/oemof/oemof-solph. You can use tespy to derive conversion efficiencies of one type of energy to mass and other types of energy (i.e. solar energy to molten salt mass flow, and molten salt mass flow to storage temperature change) and then transfer those conversion efficiencies to the oemof-solph components. With oemof-solph you can then calculate the dispatch strategy of the storage and the plant, also following non-constant load or the use of additional combustion, in case no solar energy is available in order to generate electricity at minimum cost or with minimum emissions.

Best

Francesco

**Reply by PabloCastilloFCR** [reply]
Thank you, Francesco. I will take your suggestions into account and will update you on our progress with the transient simulation of the solar thermal system. Several engineers within the company are interested in using TESPy, so we are eagerly working on this task.

Best,
Pablo
